### PR TITLE
Update CORS

### PR DESCRIPTION
**Pull request by brendansudol**
* sets origin to reflect the request origin
* now passes the `Access-Control-Allow-Credentials` header

on the front-end, we now have to explicitly set the `withCredentials` param to `true` for all axios requests. I created an axios instance that does this in `./util/api` (i wasn't sure where to put this)

### This pull request is ready to merge when...
- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented

### CODE REVIEW

File: api/endpoint-tests/auth/login.js [hunk 1]
```diff
@@ -9,6 +9,18 @@ tap.test('login endpoint | /auth/login', async loginTest => {
...
+  loginTest.test('proper response headers', async headerTest => {
+    const { response } = await request.post(url);
+    const { headers } = response;
+
+    headerTest.equal(headers.vary, 'Origin', 'vary origin');
+    headerTest.equal(
+      headers['access-control-allow-credentials'],
+      'true',
+      'allow credentials'
+    );
```
Comment (mgwalker) [inline api/endpoint-tests/auth/login.js]: Nice!  I'm curious if these two asserts should be part of all the login tests, but that's probably overkill.

File: api/main.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const routes = require('./routes');
 const server = express();
 
 server.use(express.urlencoded({ extended: true }));
-server.use(cors());
+server.use(cors({ credentials: true, origin: true }));
```
Comment (mgwalker) [inline api/main.js]: :tada:

File: web/.eslintrc.json [hunk 3]
```diff
@@ -4,6 +4,7 @@
   "rules": {
     "arrow-parens": ["error", "as-needed"],
     "comma-dangle": [1, "never"],
+    "no-confusing-arrow": 0,
```
Comment (mgwalker) [inline web/.eslintrc.json]: Why would we disable this rule?

File: web/src/util/api.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import axiosClient from 'axios';
+
+const axios = axiosClient.create({ withCredentials: true });
+
+export default axios;
```
Comment (mgwalker) [inline web/src/util/api.js]: If we don't want to wrap up a single axios instance, we can also set `withCredentials` as a default option for all axios instances:

```
axios.defaults.withCredentials = true;
```

Then we'd have to find a home for *that*, but anyway...  😛

File: web/src/containers/Login.js [hunk 5]
```diff
@@ -21,11 +21,12 @@ class Login extends Component {
...
 
   render() {
-    const { fetching, error, authenticated } = this.props;
+    const { authenticated, error, fetching, location } = this.props;
+    const { from } = location.state || { from: { pathname: '/' } };
     const { username, password } = this.state;
 
     if (authenticated) {
-      return <Redirect to="/" />;
+      return <Redirect to={from} />;
```
Comment (mgwalker) [inline web/src/containers/Login.js]: Nice!

File: web/src/util/api.js (hunk 4 above)
Comment (brendansudol) [reply web/src/util/api.js]: yeah i saw that -- wasn't sure which one was better or where its home should be either!

Comment (mgwalker) [reply web/src/util/api.js]: I'm on board with leaving it here for now and we can figure out a better home when we're making more than one call.  😄

File: api/main.js (hunk 2 above)
Comment (mgwalker) [reply api/main.js]: Now that I've thought about this a little more, and after doing some reading about what CORS is even supposed to be, I think the right solution from a security standpoint is to set the origin to a static value.  It'd probably have to come from an environment variable so we could vary it based on what was consuming it.  CORS still isn't super helpful, but 🤷‍♂️ 

Anyway, I'll open a backlog issue for that, this is perfectly fine for now.

**Pull request merged**